library/pk.c

```diff
@@ -518,6 +523,38 @@ int mbedtls_pk_sign( mbedtls_pk_context *ctx, mbedtls_md_type_t md_alg,
                                          f_rng, p_rng, NULL ) );
 }
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+/*
+ * Make a signature with options
+ */
+int mbedtls_pk_sign_ext( mbedtls_pk_type_t pk_type, mbedtls_pk_context *ctx,
+                    mbedtls_md_type_t md_alg,
+                    const unsigned char *hash, size_t hash_len,
+                    unsigned char *sig, size_t sig_size, size_t *sig_len,
+                    int (*f_rng)(void *, unsigned char *, size_t), void *p_rng )
+{
+    const mbedtls_pk_info_t *pk_info = mbedtls_pk_info_from_type( pk_type );
+    PK_VALIDATE_RET( ctx != NULL );
```

Comment:
Minor: since 3.0 those macros never do anything. We haven't gone the trouble of removing them everywhere yet, but we probably shouldn't be using them in new functions.